va/va_fei_hevc.h

```diff
@@ -0,0 +1,282 @@
+ * this buffer allocation is always based on 16x16 block even block size is indicated as 32x32 or 64x64, and buffer
+ * layout is always in 32x32 block raster scan order even block size is 16x16 or 64x64. If 32x32 block size is set,
+ * only the data in the first 16x16 block (block 0) is used for 32x32 block. If 64x64 block size is set (valid only
+ * when 64x64 LCU is supported), MV layout is still in 32x32 raster scan order, the same as 32x32 and the first 16x16
+ * block within each 32x32 block needs to have intended MV data (four 32x32 blocks will have the same MV data in the
+ * correspondent first 16x16 block). Data structure for each 16x16 block is defined as below (same as AVC except
+ * BlockSize/Reserved bits).
+ **/
+typedef struct _VAEncFEIMVPredictorHevc
+{
+    /** \brief Feference index corresponding to the entry of RefPicList0 & RefPicList1 in slice header (final reference
+     * list). Note that RefPicList0 & RefPicList1 needs to be the same for all slices.
+     * Ref0xIndex – RefPicList0; Ref1xIndex – RefPicList1; x – 0 ~ MaxNumPredictor */
+    struct {
+        uint8_t   ref_idx_l0    : 4;
+        uint8_t   ref_idx_l1    : 4;
+    } ref_idx[4]; /* index is predictor number */
+    /** \brief Valid only when MVPredictor is set to 011 for Hevc. Only valid in the first 16x16 block.
+     * 00: MV predictor disabled for this 32x32 block
+     * 01: MV predictor enabled per 16x16 block for this 32x32 block
+     * 10: MV predictor enabled per 32x32 block, the rest of 16x16 block data within this 32x32 block are ignored
+     * 11: Reserved */
+    uint32_t block_size         : 2;
+    uint32_t reserved           : 30;
+
+    VAMotionVector mv[4]; /* MaxNumPredictor is 4 */
+} VAEncFEIMVPredictorHevc;    //40 bytes
+
+/** \brief FEI CTB level control data structure */
+typedef struct _VAEncFEICTBControlHevc
+{
+    // DWORD 0
+    uint32_t    force_to_intra      : 1;
+    uint32_t    force_to_inter      : 1;
+    uint32_t    force_to_skip       : 1;
+    /** \brief force all coeff to zero */
+    uint32_t    force_to_zero_coeff : 1;
+    uint32_t    reserved0           : 28;
+    // DWORD 1
+    uint32_t    reserved1;
+    // DWORD 2
+    uint32_t    reserved2;
+    // DWORD 3
+    uint32_t    reserved3;
+} VAEncFEICTBControlHevc;
+
+/** \brief VAEncFEICTBCmdHevc defines the data structure for VAEncFEICTBCmdBufferType per CTB block.
+ * it is output buffer of ENC and ENC_PAK modes, it's also input buffer of PAK mode.
+ * The CTB block is in raster scan order. Buffer size shall not be less than the number of CTB blocks
+ * multiplied by sizeof(VAEncFEICTBCmdHevc). Note that, when separate ENC and PAK is enabled,
+ * the exact layout of this buffer is needed for PAK input. App can reuse this buffer,
+ * or copy to a different buffer as PAK input, reserved elements must not be modified when used as PAK input.
+ **/
+typedef struct _VAEncFEICTBCmdHevc
+{
+    // DWORD 0
+    uint32_t    reserved0;
```

Comment:
Could you move the reserved bytes to the end of this structure?

Reply:
No, this struct is HW PAK engine input layout, we can not move reserved0, otherwise HW PAK may gets hang.

Reply:
so this buffer is hardware dependent, right?

Reply:
yes, actually the first DWORD probable includes some debug info , which does not want customers know, it does not mean only a reserved DWORD.

Reply:
Yes